✗ Android license status unknown.

## Steps to Reproduce

> Please tell us what you were doing and what went wrong. If you are running flutter tools from the command line, please try adding the `-v` or `--verbose` option to gather more information.

1. Run `flutter upgrade`.

**What went wrong**
> **✗ Android license status unknown.**

**What should happen**
There shouldn't be a problem. The build from last week did not output a problem, and this week's build shouldn't output a problem.

**What I tried to fix the problem**
1. Upgraded Android Studio
2. Added a few random SDK in the SDK Manager
3. `sudo chown -R $USER:$USER /opt/*`
4. `flutter doctor --android-licenses`

> ~~If the problem is with your application's rendering, please attach a screenshot and any relevant source code.~~
~~If you are getting an exception in the logs, and your code is implicated in the first few frames, then please include the source code for the functions involved.~~

## Logs

> Run your application with `flutter run` and attach all the log output.

```Bash
Launching lib/main.dart on Nexus 5 in debug mode...
Initializing gradle...                                       0.8s
Resolving dependencies...                                   11.8s
Running 'gradlew assembleDebug'...                           4.7s
Built build/app/outputs/apk/debug/app-debug.apk (26.9MB).
```

The app then opens in the device and immediately crashes.

> Run `flutter analyze` and attach any output of that command also.

```Bash
Analyzing /home/users/n.gendron/GitLab/n.gendron/techno_store...
  lint • Close instances of `dart.core.Sink` at lib/server.dart:10:20 • close_sinks
  hint • 'BASE64' is deprecated and shouldn't be used at lib/edit.dart:85:9 • deprecated_member_use
  hint • 'JSON' is deprecated and shouldn't be used at lib/globals.dart:70:12 • deprecated_member_use
  hint • 'JSON' is deprecated and shouldn't be used at lib/globals.dart:80:19 • deprecated_member_use
  hint • 'BASE64' is deprecated and shouldn't be used at lib/item.dart:83:11 • deprecated_member_use
  hint • 'JSON' is deprecated and shouldn't be used at lib/server.dart:50:36 • deprecated_member_use
  hint • 'JSON' is deprecated and shouldn't be used at lib/server.dart:129:17 • deprecated_member_use
7 issues found.
(Ran in 8.2s)
```

Oh, what happened the last two days?

## Flutter Doctor

> Paste the output of running `flutter doctor -v` here.

```Bash
[✓] Flutter (Channel beta, v0.2.3, on Linux, locale en_US.UTF-8)
    • Flutter version 0.2.3 at /home/users/n.gendron/GitHub/flutter
    • Framework revision 5a58b36e36 (2 weeks ago), 2018-03-13 13:20:13 -0700
    • Engine revision e61bb9ac3a
    • Dart version 2.0.0-dev.35.flutter-290c576264

[!] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /opt/android-sdk/
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • ANDROID_HOME = /opt/android-sdk/
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    ✗ Android license status unknown.

[✓] Android Studio (version 3.1)
    • Android Studio at /opt/android-studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] Connected devices (1 available)
    • Nexus 5 • 05473ea90a673764 • android-arm • Android 7.1.2 (API 25)

! Doctor found issues in 1 category.
```

```Bash
flutter doctor --android-licenses
Warning: java.net.ConnectException: Connection timed out (Connection timed out) 
Warning: Failed to download any source lists! Fetch remote repository...        
Warning: File /home/users/n.gendron/.android/repositories.cfg could not be loaded.
All SDK package licenses accepted.======] 100% Computing updates... 
```

> For more information about diagnosing and reporting Flutter bugs, please see [https://flutter.io/bug-reports/](https://flutter.io/bug-reports/).

**Edit :** Oh hey, I forgot to add some info about my system. It should probably be noted somewhere in the template to add it.

```Bash
neofetch
            .-/+oossssoo+/-.               n.gendron@atelier02-570f 
        `:+ssssssssssssssssss+:`           ------------------------ 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 16.04.4 LTS x86_64 
    .ossssssssssssssssssdMMMNysssso.       Kernel: 4.13.0-38-generic 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Uptime: 17 hours, 31 mins 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Packages: 3069 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Shell: bash 4.3.48 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Resolution: 1024x768, 1920x1080 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   DE: GNOME 3.18.5 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: GNOME Shell 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM Theme: Adwaita 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Theme: Greybird [GTK2/3] 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Icons: Elementary-xfce-darker [GTK2/3] 
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    CPU: Intel i5 660 (4) @ 2.035GHz 
  +sssssssssdmydMMMMMMMMddddyssssssss+     GPU: NVIDIA GeForce 8400 GS Rev. 2 
   /ssssssssssshdmNNNNmyNMMMMhssssss/      Memory: 4576MiB / 7840MiB 
    .ossssssssssssssssssdMMMNysssso. 
      -+sssssssssssssssssyyyssss+-                                 
        `:+ssssssssssssssssss+:` 
            .-/+oossssoo+/-. 
```
```Bash
java -version
openjdk version "1.8.0_151"
OpenJDK Runtime Environment (build 1.8.0_151-8u151-b12-0ubuntu0.16.04.2-b12)
OpenJDK 64-Bit Server VM (build 25.151-b12, mixed mode)
```

Trying @Zedonboy's suggestion gave me some more problems...

```
flutter doctor --android-licenses
A newer version of the Android SDK is required. To update, run:
C:\Users\Caleb\AppData\Local\Android\sdk\tools\bin\sdkmanager --update
```

```
C:\Users\Caleb\AppData\Local\Android\sdk\tools\bin\sdkmanager --update
Exception in thread "main" java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlSchema
        at com.android.repository.api.SchemaModule$SchemaModuleVersion.<init>(SchemaModule.java:156)
        at com.android.repository.api.SchemaModule.<init>(SchemaModule.java:75)
        at com.android.sdklib.repository.AndroidSdkHandler.<clinit>(AndroidSdkHandler.java:81)
        at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:73)
        at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:48)
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.annotation.XmlSchema
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:582)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:190)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:499)
        ... 5 more
```

@Herohtar - if you modify the tools you need to also clear the build tools snapshot to rebuild them otherwise that one line change wont do anything. If you read the link I posted the workaround is to use the `SDKMANAGER_OPTS` environment variable. That's all that my change is actually doing.

On a mac it would be:

```
export SDKMANAGER_OPTS="--add-modules java.se.ee"
```

This should solve the `Caused by: java.lang.ClassNotFoundException: javax.xml.bind.annotation.XmlSchema` error that you are seeing when `sdkmanager` is run.

Alternatively, that link offers other suggestions. If you find out that the solution is different, report back.

The link again in case you missed it: https://stackoverflow.com/questions/47150410/failed-to-run-sdkmanager-list-android-sdk-with-java-9.

--- a/packages/flutter_tools/lib/src/android/android_sdk.dart
+++ b/packages/flutter_tools/lib/src/android/android_sdk.dart
@@ -332,6 +332,7 @@ class AndroidSdk {
     if (_sdkManagerEnv == null) {
       // If we can locate Java, then add it to the path used to run the Android SDK manager.
       _sdkManagerEnv = <String, String>{};
+      _sdkManagerEnv['SDKMANAGER_OPTS'] = '--add-modules java.se.ee';
       final String javaBinary = findJavaBinary();
       if (javaBinary != null) {
         _sdkManagerEnv['PATH'] =

